Examples/ObjectiveC/ClusteringExample.m

```diff
@@ -0,0 +1,138 @@
+                             @50:  [MGLStyleValue valueWithRawValue:[UIColor orangeColor]],
+                             @100: [MGLStyleValue valueWithRawValue:[UIColor redColor]],
+                             @200: [MGLStyleValue valueWithRawValue:[UIColor purpleColor]] };
+
+    // Show clustered features as circles. The `point_count` attribute is built into clustering-enabled source features.
+    MGLCircleStyleLayer *circlesLayer = [[MGLCircleStyleLayer alloc] initWithIdentifier:@"clusteredPorts" source:source];
+    circlesLayer.circleRadius = [MGLStyleValue valueWithRawValue:@(self.sprite.size.width / 2)];
+    circlesLayer.circleOpacity = [MGLStyleValue valueWithRawValue:@0.75];
+    circlesLayer.circleStrokeColor = [MGLStyleValue valueWithRawValue:[[UIColor whiteColor] colorWithAlphaComponent:0.75]];
+    circlesLayer.circleStrokeWidth = [MGLStyleValue valueWithRawValue:@2];
+    circlesLayer.circleColor = [MGLSourceStyleFunction functionWithInterpolationMode:MGLInterpolationModeInterval
+                                                                              stops:stops
+                                                                      attributeName:@"point_count"
+                                                                            options:nil];
+    circlesLayer.predicate = [NSPredicate predicateWithFormat:@"%K == YES", @"cluster"];
+    [style addLayer:circlesLayer];
+
+    // Label cluster circles with a layer of text indicating feature count. Per text token convention, wrap the attibute in {}.
+    MGLSymbolStyleLayer *numbersLayer = [[MGLSymbolStyleLayer alloc] initWithIdentifier:@"clusteredPortsNumbers" source:source];
+    numbersLayer.textColor = [MGLStyleValue valueWithRawValue:[UIColor whiteColor]];
+    numbersLayer.textFontSize = [MGLStyleValue valueWithRawValue:@(self.sprite.size.width / 2)];
+    numbersLayer.iconAllowsOverlap = [MGLStyleValue valueWithRawValue:@(YES)];
+    numbersLayer.text = [MGLStyleValue valueWithRawValue:@"{point_count}"];
+    numbersLayer.predicate = [NSPredicate predicateWithFormat:@"%K == YES", @"cluster"];
+    [style addLayer:numbersLayer];
+
+    // Add a tap gesture for zooming in to clusters or showing popups on individual features.
+    [self.view addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleTap:)]];
+}
+
+- (void)mapViewRegionIsChanging:(MGLMapView *)mapView {
+    [self showPopup:NO animated:NO];
+}
+
+- (void)handleTap:(UITapGestureRecognizer *)tap {
+    if (tap.state == UIGestureRecognizerStateEnded) {
+        CGPoint point = [tap locationInView:tap.view];
+        CGFloat width = self.sprite.size.width;
+        CGRect rect = CGRectMake(point.x - width / 2, point.y - width / 2, width, width);
+
+        NSArray *clusters = [self.mapView visibleFeaturesInRect:rect inStyleLayersWithIdentifiers:[NSSet setWithObject:@"clusteredPorts"]];
```

Comment:
Let’s add some comments about what we’re doing here, as `visibleFeaturesInRect: inStyleLayersWithIdentifiers:` is somewhat specialized.